process-dataset/data-validation/schema.js

```diff
@@ -0,0 +1,261 @@
+        },
+        "key": {
+            "description": "Id of the feature",
+            "type": "string",
+        },
+        "discrete": {
+            "description": "Whether it's a continuous measurement or not",
+            "type": "boolean",
+        },
+        "options": {
+            "description": "For discrete features, display items for each value",
+            "type": "object",
+        },
+
+    },
+    "required": [
+        "displayName",
+        "description",
+        "tooltip",
+        "unit",
+        "key",
+        "discrete",
+    ],
+}
+
+const fileInfoSchema = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "fileInfoSchema",
+    "title": "File Info",
+    "description": "Metadata per cell",
+    "type": "array",
+    "items": {
+        "additionalProperties": false,
+        "type": "object",
+        "properties": {
+            "CellId": {
+                "description": "unique id for segmented cell",
+                "type": "number"
+            },
+            "FOVId": {
+                "description": "Field of view cell came from",
+                "type": "number",
+            },
+            "CellLineName": {
+                "description": "Id of cell line",
+                "type": "string",
+            },
+            "thumbnailPath": {
+                "description": "Path to thumbnail image for cell",
+                "type": "string",
+                "pattern": "(.)+(\.png)"
+            },
+            "volumeviewerPath": {
+                "description": "path to 3d data for cell",
+                "type": "string",
+                "pattern": "(.)+\.json"
+            },
+            "fovThumbnailPath": {
+                "description": "Path to fov thumbnail",
+                "type": "string",
+                "pattern": "(.)+\.png"
+            },
+            "fovVolumeviewerPath": {
+                "description": "Path to the fov 3d data",
+                "type": "string",
+                "pattern": "(.)+\.json"
+            },
+
+        },
+        "required": [
+            "CellId",
+            "FOVId",
+            "CellLineName",
+            "thumbnailPath",
+            "volumeviewerPath",
+            "fovThumbnailPath",
+            "fovVolumeviewerPath"
+        ],
+    }
+}
+
+const measuredFeaturesDocSchema = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "measuredFeaturesSchema",
+    "title": "Measured features",
+    "description": "Data per cell needed to create the CFE plot",
+    "type": "object",
+    "type": "array",
+    "items": {
+        "additionalProperties": false,
+        "type": "object",
+        "properties": {
+            "f": {
+                "description": "ordered array of measured features",
+                "type": "array",
+                "items": {
+                    "type": "number"
+                }
+            },
+            "p": {
+                "description": "Protein name",
```

Comment:
I wonder if cell line name would be a more abbreviated way of delivering this info, since the app should have all the cell line data.. maybe not a huge difference tho

Reply:
This is the actual value we use on the plot, and this document is basically the min data needed to create the plot, this made sense to me to have here.